ptable/ptable.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Partition table
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#ifndef _LIB_PARTITION_TABLE_H_
+#define _LIB_PARTITION_TABLE_H_
+
+
+/* Structure of partition table:
+*  _________________________________________________________________________________
+* |        28 B     |                      32 B * n                     |    4 B    |
+*  ---------------------------------------------------------------------------------
+* | ptable_header_t | ptable_partition_t 0 | ... | ptable_partition_t n | magicBytes|
+*  ---------------------------------------------------------------------------------
+*
+*    NOTE: Data in the partition table should be stored in little endian.
+*
+*/
+
+
+#include <stdint.h>
+
+
+static const uint8_t pt_magicBytes[] = {0xde, 0xad, 0xfc, 0xbe};
+
+/* Based on MBR types */
+enum { ptable_raw = 0x51, ptable_meterfs = 0x75 };
+
+
+typedef struct ptable_partition {
+	unsigned char name[8];
+
+	uint32_t accessMode;
+
+	uint32_t offset;
+	uint32_t size;
+
+	uint8_t type;
+	uint8_t reserved[11];     /* default value should be 0 */
+} __attribute__((packed)) ptable_partition_t;
+
+
+typedef struct ptable_header {
+	uint32_t pCnt;            /* number of partitions */
+	uint8_t reserved[24];     /* default value should be 0 */
+
+	ptable_partition_t parts[0];
+} __attribute__((packed)) ptable_header_t;
+
+
+typedef struct memory_properties {
+	uint32_t memSize;
+	uint32_t sectorSize;
+
+	size_t (*read)(unsigned int addr, void *buff, size_t bufflen);
+} memory_properties_t;
+
+
+
+/* Function check correctness of the partition's parameters */
+int ptable_verifyPartition(uint8_t id, const ptable_partition_t *pHeaders, const memory_properties_t *mem);
+
+
+/* Function allocates pHeaders - pointer to an array of partitions and  writes to pHeadersLen number of partitions. */
+int ptable_getPartitions(ptable_partition_t **pHeaders, uint8_t *pHeadersLen, const memory_properties_t *mem);
```

Comment:
This is actually `readPartitons` (as it's doing `read` operations, there will be `writePartitions` counterpart in the future)

Why noting is being returned from function the standard way? I think the standard way is that if the function allocate the memory, it returns it directly (not via output parameter)... but the function could have read-alike interface (returns number of partitions read or < 0 on error)